e2e/e2e-helper.sh

```diff
@@ -16,6 +16,97 @@ exec_on_host() {
+    az storage container create -n "tmp" --account-name $WINDOWS_E2E_STORAGE_ACCOUNT --account-key $account_key
+    set -x
+}
+
+upload_linux_file_to_storage_account() {
+    local retval=0
+    MC_RESOURCE_GROUP_NAME="MC_${RESOURCE_GROUP_NAME}_${CLUSTER_NAME}_$LOCATION"
+    MC_VMSS_NAME=$(az vmss list -g $MC_RESOURCE_GROUP_NAME --query "[?contains(name, 'nodepool')]" -ojson | jq -r '.[0].name')
+    VMSS_INSTANCE_ID="$(az vmss list-instances --name $MC_VMSS_NAME -g $MC_RESOURCE_GROUP_NAME | jq -r '.[0].instanceId')"
+
+    set +x
+    expiryTime=$(date --date="2 day" +%Y-%m-%d)
+    account_key=$(az storage account keys list --account-name $WINDOWS_E2E_STORAGE_ACCOUNT --query "[0].value")
+    token=$(az storage container generate-sas --account-name $WINDOWS_E2E_STORAGE_ACCOUNT --account-key $account_key --permissions 'rwacdl' --expiry $expiryTime --name "tmp")
+    linuxFileURL="https://${WINDOWS_E2E_STORAGE_ACCOUNT}.blob.core.windows.net/tmp/${MC_VMSS_NAME}-linux-file.zip?${token}"
+
+    az vmss run-command invoke --command-id RunShellScript \
+        --resource-group $MC_RESOURCE_GROUP_NAME \
+        --name $MC_VMSS_NAME \
+        --instance-id $VMSS_INSTANCE_ID \
+        --scripts "cat /etc/kubernetes/azure.json > /home/fields.json; cat /etc/kubernetes/certs/apiserver.crt | base64 -w 0 > /home/apiserver.crt; cat /etc/kubernetes/certs/ca.crt | base64 -w 0 > /home/ca.crt; cat /etc/kubernetes/certs/client.key | base64 -w 0 > /home/client.key; cat /var/lib/kubelet/bootstrap-kubeconfig > /home/bootstrap-kubeconfig; cd /home; zip file.zip fields.json apiserver.crt ca.crt client.key bootstrap-kubeconfig; wget https://aka.ms/downloadazcopy-v10-linux; tar -xvf downloadazcopy-v10-linux; cd ./azcopy_*; ./azcopy copy /home/file.zip $linuxFileURL" || retval=$?
+    
+    set -x
+    if [ "$retval" -eq 0 ]; then
+        log "upload linux file successfully"
+    else
+        err "failed to upload linux file"
+    fi
+}
+
+download_linux_file_from_storage_account() {
+    wget https://aka.ms/downloadazcopy-v10-linux
+    tar -xvf downloadazcopy-v10-linux
+
+    expiryTime=$(date --date="2 day" +%Y-%m-%d)
+    for i in $(seq 1 20); do
+        accountState=$(az storage account list -g $RESOURCE_GROUP_NAME | jq '.[0].provisioningState' | tr -d "\"")
+        if [ "$accountState" != "Succeeded" ]; then
+            log "Other pipeline may be creating the storage account, waiting..."
+            sleep 10
+            continue
+        fi
+        break;
+    done
+
+    set +x
+
+    local retval=0
+    for i in $(seq 1 20); do
+        account_key=$(az storage account keys list --account-name $WINDOWS_E2E_STORAGE_ACCOUNT --query "[0].value") || retval=$?
+        if [ "$retval" -ne 0 ]; then
+            sleep 10
+            continue
+        fi
+        break;
+    done
+    token=$(az storage container generate-sas --account-name $WINDOWS_E2E_STORAGE_ACCOUNT --account-key $account_key --permissions 'rwacdl' --expiry $expiryTime --name "tmp")
```

Comment:
can we only grant 'r' to download the file?

Reply:
Will test permissions needed and change code.